span_test.go

```diff
@@ -168,13 +173,62 @@ var spanPairs = map[string]struct {
 			Meta:    map[string]string{},
 		},
 	},
+	"globalTags": {
+		options: Options{
+			Service: "my-service",
+			GlobalTags: map[string]interface{}{
+				"key1": "value1",
+				"key2": false,
+				"key3": int64(2),
+			},
+		},
+		oc: &trace.SpanData{
+			SpanContext: trace.SpanContext{
+				TraceID:      trace.TraceID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}),
+				SpanID:       trace.SpanID([8]byte{1, 2, 3, 4, 5, 6, 7, 8}),
+				TraceOptions: 1,
+			},
+			SpanKind:  trace.SpanKindServer,
+			Name:      "/a/b",
+			StartTime: testStartTime,
+			EndTime:   testEndTime,
+			Attributes: map[string]interface{}{
+				"str":   "abc",
+				"bool":  true,
+				"int64": int64(1),
+			},
+			Status: trace.Status{},
+		},
+		dd: &ddSpan{
+			TraceID:  651345242494996240,
+			SpanID:   72623859790382856,
+			Type:     "server",
+			Name:     "/a/b",
+			Resource: "/a/b",
+			Start:    testStartTime.UnixNano(),
+			Duration: testEndTime.UnixNano() - testStartTime.UnixNano(),
+			Metrics: map[string]float64{
+				"key3":              2,
+				"int64":             1,
+				samplingPriorityKey: ext.PriorityAutoKeep,
+			},
+			Service: "my-service",
+			Meta: map[string]string{
+				"key1": "value1",
+				"key2": "false",
+				"bool": "true",
+				"str":  "abc",
+			},
+		},
+	},
 }
 
 func TestConvertSpan(t *testing.T) {
-	service := "my-service"
-	e := newTraceExporter(Options{Service: service})
 	for name, tt := range spanPairs {
```

Comment:
We're in a tight spot because you're trying to fit in the tests in the wrong place. Just create a new test:
```go
func TestGlobalTags(t *testing.T) {
    e := newTraceExporter(Options{
        Service: "my-service",
        GlobalTags: map[string]interface{}{"key1": "value1"},
    })
    defer e.stop()

    got := e.convertSpan(spanPairs["tags"].oc)
    if got.Meta["key1"] != "value1" {
        t.Fatal("global tag not set")
    }
}
```
Should be more than enough.

Reply:
I see. I understood your previous advice!.